src/libraries/System.Private.DataContractSerialization/src/System/Xml/EncodingStreamWrapper.cs

```diff
@@ -426,15 +426,8 @@ private static bool CompareCaseInsensitive(byte[] key, byte[] buffer, int offset
             return true;
         }
 
-        private static bool Compare(byte[] key, byte[] buffer, int offset)
-        {
-            for (int i = 0; i < key.Length; i++)
-            {
-                if (key[i] != buffer[offset + i])
-                    return false;
-            }
-            return true;
-        }
+        private static bool Compare(byte[] key, byte[] buffer, int offset) =>
```

Comment:
Is the buffer guaranteed to have at least `key.Length` elements?

It may be better to inline this into the one callsite where this is used.